app/scripts/controllers/buildConfig.js

```diff
@@ -65,6 +66,40 @@ angular.module('openshiftConsole')
               });
             }
 
+            $scope.updatedBuildConfig = angular.copy($scope.buildConfig);
+            $scope.buildStrategy = $filter('buildStrategy')($scope.updatedBuildConfig);
```

Comment:
dont have a local variable and a scope variable that use the same name and are not pointing to the same thing